src/btree/row_modify.c

```diff
@@ -334,7 +334,7 @@ __wt_update_obsolete_check(
      * Only updates with globally visible, self-contained data can terminate update chains.
      *
      */
-    for (first = NULL, count = 0; upd != NULL; upd = upd->next, count++) {
+    for (first = NULL; upd != NULL; upd = upd->next, (*upd_count)++) {
```

Comment:
We should try to protect __wt_update_obsolete_check if called with a NULL upd_count. It should be able to function if a NULL is passed. We can keep and use the local "count" variable, and if the caller has passed a non NULL "upd_count", we can populate it with the update count. This sort of usage is common in the WiredTiger code.

As an example look at __wt_meta_ckptlist_get_from_config(). It takes an optional "size_t *allocatedp". It uses a local variable, "allocated" for the needed functionality, but at the end copies the result into "allocatedp" if not NULL.

Reply:
I think we're counting updates that we're going to discard in this loop, that is, we could have a long list of updates that we set the `WT_UPDATE *` return and then discard, but still evict the page. Honestly, it's hard to imagine that would make a difference, but thought I'd point it out.